src/table.rs

```diff
@@ -45,74 +65,14 @@ impl Drop for TableInner {
         fs::remove_file(&f.path).unwrap();
     }
 }
+*/
 
+#[derive(Default)]
 pub struct Block {
     offset: usize,
     data: Bytes,
-    checksum: Vec<u8>,
-    checksum: u64,
+    checksum: Bytes,
     entries_index_start: usize,
     entry_offsets: Vec<u32>,
     checksum_len: usize,
 }
-
-impl Block {
-    fn size(&self) -> u64 {
-        3 * mem::size_of::<usize>() as u64 + self.data.len() as u64 + self.checksum.len() as u64 + self.entry_offsets.len() as u64 * mem::size_of::<u32>() as u64
-    }
-
-    fn verify_checksum(&self) -> Result<()> {
-        let mut chksum = CheckSum::default();
-        chksum.merge_from_bytes(&self.checksum)?;
-        checksum::verify_checksum(&self.data, &chksum)
-    }
-}
-
-fn parse_file_id(name: &str) -> (u64, bool) {
-    if !name.ends_with(".sst") {
-        (0, false)
-    }
-    match name[..name.len() - 4].parse() {
-        Ok(id) => (id, true),
-        Err(_) => return (0, false),
-    }
-}
-
-impl Table {
-    pub fn open(path: &Path, opt: Options) -> Result<Table> {
```

Comment:
Why delete it?

Reply:
I've added them back. In fact this Table implementation differs a lot from the final one, I think we could compare them in next few PRs.